File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Tanimlamalar/CityColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CityRow } from "./CityRow";

export interface CityColumns {
    CityId: Column<CityRow>;
    Name: Column<CityRow>;
}

export class CityColumns extends ColumnsBase<CityRow> {
    static readonly columnsKey = 'Tanimlamalar.City';
    static readonly Fields = fieldsProxy<CityColumns>();
}

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Tanimlamalar/City/CityColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace MuayeneYonetimPortali.Tanimlamalar.Columns;

[ColumnsScript("Tanimlamalar.City")]
[BasedOnRow(typeof(CityRow), CheckNames = true)]
public class CityColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int CityId { get; set; }
    [EditLink]
    public string Name { get; set; }
}